### ISSUE TITLE
Cycle inside Alamofire watchOS - Error in Xcode 13.3 when using new build system

### ISSUE THREAD

**Issue by zzt4** [Alamofire/Alamofire#3588]
* [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## What did you do?

I created a new watchOS framework in Xcode with Alamofire as a dependency.

## What did you expect to happen?

A successful build of the framework.

## What happened instead?

The build failed with this error message: 

```
Cycle inside Alamofire watchOS; building could produce unreliable results. This usually can be resolved by moving the target's Headers build phase before Compile Sources.
```

## Alamofire Environment

**Alamofire Version:** 5.4.4 _and_ 5.5.0
**Dependency Manager:** Carthage
**Xcode Version:** Xcode 13.3
**Swift Version:** Swift 5.6
**Platform(s) Running Alamofire:** iOS, watchOS
**macOS Version Running Xcode:** macOS Monterey 12.3

## Demo Project

Cannot provide demo project due to open source rules at our company.

This does not happen every time when building. Sometimes after a fresh clean, and sometimes after successful builds it might happen a few builds later. We have also noticed this with our internal frameworks and moving the "headers" build phase above "compile sources" seems to fix it for our frameworks as well. I think this is happening for the iOS framework as well.

**Comment by jshier** [comment]
Thanks for the report. I've fixed this in the `update/ci` branch, give it a try if you could. Turns out there's nothing in those header phases so I just deleted them.

**Comment by jshier** [comment]
This may also be triggered by the new Swift build system integration in Xcode 13.2+, so you can try disabling that and see if it still happens.

**Comment by zzt4** [comment]
We don't get this issue in Xcode 13.2.1 or if we switch back to the legacy build system, so it seems to only be in Xcode 13.3 with the new build system. I'm guessing they added a new check in 13.3 to catch a potential problem with the new build system.

**Comment by jshier** [comment]
This has been merged. You can use the `master` branch until we release it as part of Alamofire 5.6. Thanks again!

**Comment by funcJeff** [comment]
@jshier my project cannot use anything past AF 5.5 right now. Can we get a 5.4.5 release with this change?

**Comment by jshier** [comment]
There is nothing past Alamofire 5.5 yet. This fix will roll out as part of 5.6 soon.
